src/main/java/raisetech/student/management2/controller/StudentController.java

```diff
@@ -56,16 +65,19 @@ public List<StudentDetail>  getStudentList() {
 
   /**
    * 受講生詳細検索を行う
-   * IDに紐づくに似の受講生の情報を取得する
+   * IDに紐づく受講生の情報を取得する
    * @param id 受講生ID
    * @return 受講生詳細
    */
   @GetMapping("/student/{id}")
-  public StudentDetail getStudent(@PathVariable @Size(min=1, max=2, message="入力して！！！") String id) {
-
-    //studentDetail.setStudentsCourses(Arrays.asList(new StudentsCourse()));
+  public StudentDetail getStudent(@PathVariable(required = false) String id) {
+    if (id == null || id.trim().isEmpty() || !id.matches("\\d{1,2}")) {
+      System.out.println("受講生IDが入力されていません。");
+      throw new StudentNotFoundException("受講生IDが存在しなません");
```

Comment:
- IDが入力されていないこと
- IDに紐づく受講生が存在しないこと

これらは全く別の事象ですね。
`System.out.prinln()`は消しておきましょう。

また、`受講生IDが存在しなません`という日本語もおかしそうです

```suggestion
      throw new StudentNotFoundException("IDに紐づく受講生が存在しなません");
```